src/app/register/register.component.html

```diff
@@ -0,0 +1,139 @@
+      label="Last Name"
+      [control]="getControlName(registrationForm, 'lastName')"
+    >
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('lastName')?.touched && registrationForm.get('lastName')?.errors?.['required']"
+      >
+        *Last name is required
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('lastName')?.touched && registrationForm.get('lastName')?.errors?.['minlength']"
+      >
+        *Please enter at least 3 character
+      </small>
+    </app-input>
+
+    <app-input
+      placeholder="Enter your user name"
+      type="text"
+      label="User Name"
+      [control]="getControlName(registrationForm, 'userName')"
+    >
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('userName')?.touched && registrationForm.get('userName')?.errors?.['required']"
+      >
+        *User name is required
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('userName')?.touched && registrationForm.get('userName')?.errors?.['uniqueUserNameError']"
+      >
+        *This user is already registered
+      </small>
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('userName')?.touched && registrationForm.get('userName')?.errors?.['noSpaceAllowed']"
+      >
+        *No space allowed.
+      </small>
+    </app-input>
+
+    <app-input
+      placeholder="Enter your email"
+      type="email"
+      label="Email"
+      [control]="getControlName(registrationForm, 'email')"
+    >
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('email')?.touched && registrationForm.get('email')?.errors?.['required']"
+      >
+        *Email is required
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('email')?.touched && registrationForm.get('email')?.errors?.['email']"
+      >
+        *Enter a valid email
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('email')?.touched && registrationForm.get('email')?.errors?.['uniqueEmailError']"
+      >
+        *This email id is already taken
+      </small>
+    </app-input>
+    <app-input
+      placeholder="Enter your password"
+      type="password"
+      label="Password"
+      [control]="getControlName(registrationForm, 'password')"
+    >
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('password')?.touched && registrationForm.get('password')?.errors?.['required']"
+      >
+        *Password is required
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('password')?.touched && registrationForm.get('password')?.errors?.['minlength']"
+      >
+        *Please enter at least 6 character
+      </small>
+    </app-input>
+
+    <br /><br />
```

Comment:
we don't use <br/> for spacing. rather try to achieve this behaviour using semantic html and styling